lib/util/task.js

```diff
@@ -237,10 +248,10 @@
       async = true;
       // The returned function should execute asynchronously in case
       // someone tries to do this.async()(); inside a task (WTF).
-      return function(success) {
+      return this._callOnlyOnce(function(success) {
         setTimeout(function() { complete(success); }, 1);
-      };
-    };
+      });
+    }.bind(this);
```

Comment:
Is there a specific reason for having `_callOnlyOnce` on the `Task` prototype? It isn't really related to `Task` at all. If you move it out you could skip the `.bind(this)`...

Also, have you considered using [once](https://www.npmjs.com/package/once) instead of rolling your own?